OCPBUGS-14116: Improving registries.conf explanation (#60878)
openshift/openshift-docs: modules/ztp-configuring-the-cluster-for-a-disconnected-environment.adoc

apiVersion: v1
kind: ConfigMap
metadata:
  name: assisted-installer-config-map
  namespace: "<infrastructure-operator-namespace>" <1>
  labels:
    app: assisted-service
data:
  ca-bundle.crt: | <2>
    -----BEGIN CERTIFICATE-----
    certificate contents
    -----END CERTIFICATE-----

  registries.conf: | <3>
    unqualified-search-registries = ["registry.access.redhat.com", "docker.io"]

    [[registry]]
       prefix = ""
       location = "quay.io/example-repo" <4>
       mirror-by-digest-only = true

       [[registry.mirror]]
       location = "mirror1.registry.corp.com:5000/example-repo" <5>
----
<1> The `ConfigMap` namespace must be the same as the namespace of the Infrastructure Operator.

> <2> The mirror registry’s certificate used when creating the mirror registry.

Reviewer: ```suggestion
<2> The mirror registry’s certificate that is used when creating the mirror registry.
```

> <3> The configuration file for the mirror registry. The mirror registry configuration adds mirror information to `/etc/containers/registries.conf` in the Discovery image. The mirror information is stored in the `imageContentSources` section of the `install-config.yaml` file when passed to the installation program. The Assisted Service pod running on the hub cluster fetches the container images from the configured mirror registry.

Reviewer: ```suggestion
<3> The configuration file for the mirror registry. The mirror registry configuration adds mirror information to the `/etc/containers/registries.conf` directory in the Discovery image. The mirror information is stored in the `imageContentSources` section of the `install-config.yaml` file when the information is passed to the installation program. The Assisted Service pod that runs on the hub cluster fetches the container images from the configured mirror registry.
```
- Is `/etc/containers/registries.conf` a file, a directory, or something else?
- Please check my revision for accuracy. The original is a little unclear about what is being passed to the installation program. I assumed it was the mirror information.

<4> The URL of the mirror registry. You need to use the URL from the `imageContentSources` section by running the `oc adm release mirror` command when you configure the mirror registry. For more information, see the _Mirroring the OpenShift Container Platform image repository_ section.

> <5> The registries defined in the `registries.conf` must be scoped by repository, not by registry. In this example, both the `quay.io/example-repo` and the `mirror1.registry.corp.com:5000/example-repo` are scoped by `example-repo`.

Reviewer: ```suggestion
<5> The registries defined in the `registries.conf` file must be scoped by repository, not by registry. In this example, both the `quay.io/example-repo` repository and the `mirror1.registry.corp.com:5000/example-repo` repository are scoped by the `example-repo` repository.
```
- I'm not sure if "file" and "repository" are the right nouns to use here, so please feel free to replace them with whatever makes sense.

kind: ConfigMap
metadata:
  name: assisted-installer-mirror-config
  namespace: assisted-installer <1>
  labels:
    app: assisted-service
data:
  ca-bundle.crt: <certificate> <2>
  registries.conf: | <3>
    unqualified-search-registries = ["registry.access.redhat.com", "docker.io"]

    [[registry]]
      location = <mirror_registry_url>  <4>
      insecure = false
      mirror-by-digest-only = true
----

> <1> The `ConfigMap` namespace must be the same as the namespace of the infrastructure Operator.

Reviewer 2: Capitalize i in infrastructure
